OrbitQt/OverlayWidget.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OverlayWidget.h"
+
+namespace OrbitQt {
+
+OverlayWidget::OverlayWidget(QWidget* parent)
+    : QWidget(parent), parent_(parent), ui_(std::make_unique<Ui::OverlayWidget>()) {
+  CHECK(parent != nullptr);
+  parent->installEventFilter(this);
+  ui_->setupUi(this);
+  ui_->cancelButton->setEnabled(true);
+}
+
+void OverlayWidget::paintEvent(QPaintEvent* /*event*/) {
+  QPainter(this).fillRect(rect(), {100, 100, 100, 128});
+}
+
+bool OverlayWidget::eventFilter(QObject* obj, QEvent* event) {
+  if (!obj->isWidgetType()) return false;
+
+  auto* widget = dynamic_cast<QWidget*>(obj);
+  CHECK(widget == parent_);
+
+  if (event->type() == QEvent::Resize) {
+    resize(parent_->size());
+  }
+  return false;
```

Comment:
This feels a bit confusing to me. This never filters anything, does it? So this function just acts as a hooking point, doesn't it?

What about:

```
if (obj == parent() && event->type() == QEvent::Resize) {
  resize(parentWidget()->size());
}

return QWidget::eventFilter(obj, event);
```

Reply:
Yes, it does not filter anything, but just resizes when the parent resizes. I changed the code to almost what you suggested, ptal

Reply:
Looks good!